### PR TITLE
WIP: Automation broken storage

### PR DESCRIPTION
**Pull request by maggu2810**

### CODE REVIEW

**Comment by kaikreuzer** [comment]
I will need some time and a stable Internet connection to review this, both is something I do not have right now...
But at a first glance, this seems to revert a lot of what has been done in https://github.com/eclipse/smarthome/pull/4468 - mainly having the entities strictly as interfaces, which was one of the huge gains of that work.

Wrt https://github.com/eclipse/smarthome/pull/4468#issuecomment-403564760: Yes, I had tested it and also the unit tests (which load rules from JSON files as well) have passed. Note that the de-serialisation is NOT done to the interfaces (where gson would clearly fail), but to the DTOs, which is their main reason for existence. I am not sure why you came across problems with that - maybe something of my work got lost when Ani updated the PR prior to the merge? Will need to investigate...

So in short: I would actually hope that we do not need this PR (and all its structural changes).

**Comment by kaikreuzer** [comment]
> maybe something of my work got lost when Ani updated the PR prior to the merge? Will need to investigate...

No, it is actually there (https://github.com/eclipse/smarthome/pull/4468/commits/793837fbb3ec391380ea0697a73471fc8480f73d), but I just noticed that your issue was about serialisation, not deserialisation. I'll comment on #5884...

**Comment by maggu2810** [comment]
It is about deserialization of all previously serialized rules.
I missed the point that DTOs are used now, but the problem about restoring old rules from mapdb remains.
Let's continue on #5884.
